From: Chris Heywood <gitlab@cheywood.au>
Date: Sun, 21 Aug 2022 13:55:46 +1000
Subject: messageList: Allow dismissal via backspace

Currently only the delete key can be used to dismiss notifications.
Some laptops don't have delete keys, resulting in needing to hit a
modifier to trigger delete.

Plus, on those keyboards that do have delete keys they're often a
little harder to strike than backspace.

https://gitlab.gnome.org/gnome/gnome-shell/-/issues/5789

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2435>
(cherry picked from commit f119e5282b38adee502565d5fa7a5a2042bd1792)
---
 js/ui/messageList.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/js/ui/messageList.js b/js/ui/messageList.js
index 671760b..5245b3f 100644
--- a/js/ui/messageList.js
+++ b/js/ui/messageList.js
@@ -543,7 +543,8 @@ export const Message = GObject.registerClass({
         let keysym = event.get_key_symbol();
 
         if (keysym === Clutter.KEY_Delete ||
-            keysym === Clutter.KEY_KP_Delete) {
+            keysym === Clutter.KEY_KP_Delete ||
+            keysym === Clutter.KEY_BackSpace) {
             if (this.canClose()) {
                 this.close();
                 return Clutter.EVENT_STOP;
